libraries/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts

```diff
@@ -414,8 +416,8 @@ export class WorkspaceInstallManager extends BaseInstallManager {
     return packageExtensionsChecksum;
   }
 
-  protected canSkipInstall(lastModifiedDate: Date, subspace: Subspace): boolean {
-    if (!super.canSkipInstall(lastModifiedDate, subspace)) {
+  protected canSkipInstall(lastModifiedDate: Date, subspace: Subspace, variant: string | undefined): boolean {
```

Comment:
```suggestion
  protected canSkipInstall(lastModifiedDate: Date, subspace: Subspace, variant?: string): boolean {
```

Reply:
And that goes for most places where the variant is the last argument/member of a type and uses an explicit `string | undefined` since `?` is more backwards compatible

Reply:
For internal methods, `variant?: string` is more error-prone than `variant: string | undefined`: it is easy to forget to include the `variant` parameter, and unless you are explicitly testing variants, you would not even realize that the variant feature was broken.